lib/metamocha.js

```diff
@@ -0,0 +1,127 @@
+var Mocha = require('mocha'),
+    Context = require('mocha').Context,
+    metadataUi = require('./metadata_ui'), // eslint-disable-line no-unused-vars
+    path = require('path'),
+    recursiveReadSync = require('recursive-readdir-sync'),
+    fs = require('fs');
+
+/**
+ * Metamocha Test Runner
+ */
+var Metamocha = function() {
+  this.mocha = new Mocha({ui: 'metadata_ui'});
+  this.files = [];
+  this.filters = [];
+};
+
+/**
+ * Load files and generate Test objects
+ */
+Metamocha.prototype.loadFiles = function() {
+  var self = this;
+
+  this.files.forEach(function(file) {
+    self.mocha.addFile(file);
+  });
+  this.mocha.loadFiles();
+  if (this.filters.length) {
+    this.applyFilters();
+  }
+  return this;
+};
+
+/**
+ * Add single file to the list of test files 
+ */
+Metamocha.prototype.addFile = function(file) {
+  this.files.push(file);
+  return this;
+};
+
+/**
+ * Add files contained in one directory to the list of test files
+ */
+Metamocha.prototype.addFolder = function(dir) {
+  var self = this;
+
+  if (fs.lstatSync(dir).isDirectory()) {
```

Comment:
Could you re-structure this block to bail early and thereby remove a layer of indentation? Something like:
```javascript
if (!fs.lstatSync(dir).isDirectory()) {
  throw new Error('Input must be a directory');
}

fs.readdirSync(dir).filter(function(file) {
  return file.substr(-3) === '.js';
}).forEach(function(file) {
  self.files.push(path.join(dir, file));
});
```